tamr_client/backup.py

```diff
@@ -0,0 +1,120 @@
+from typing import List
+
+from tamr_client import Backup, response
+from tamr_client._types import Instance, JsonDict, Session, URL
+from tamr_client.exception import TamrClientException
+
+
+class InvalidOperation(TamrClientException):
+    """Raised when attempting an invalid operation.
+    """
+
+    pass
+
+
+class NotFound(TamrClientException):
+    """Raised when referencing a backup that does not exist on the server.
+    """
+
+    pass
+
+
+def _from_json(data: JsonDict) -> Backup:
+    """Make backup from JSON data (deserialize).
+
+    Args:
+        data: Backup JSON data from Tamr server
+    """
+    return Backup(
+        url=data["id"],
+        resource_id=data["relativeId"],
+        path=data["backupPath"],
+        state=data["state"],
+        error_message=data["errorMessage"],
+    )
+
+
+def get_all(session: Session, instance: Instance) -> List[Backup]:
+    """Get information on Tamr backups.
```

Comment:
I think this line in the docstring (and in `from_resource_id`) could be made more specific.  If we are using the terminology `Backup` for the job/process/operation that backs up an instance of Tamr, this should instead say
```python
Get all backups that have been initiated for a Tamr instance
```